core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -933,6 +953,17 @@ class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends
     createResponse.resultException.foreach(e => throw e)
   }
 
+  private def logDirFailureEventZkData(brokerId: Int): String = {
+    val LogDirFailureEvent = 1
+    Json.encode(Map("version" -> LogDirEventNotificationHandler.Version, "broker" -> brokerId, "event" -> LogDirFailureEvent))
+  }
+
+  def propagateLogDirEvent(brokerId: Int) {
```

Comment:
hmm..This method should be like createConfigChangeNotification, createAclChangeNotification methods.
and should use existing ZkData.LogDirEventNotificationSequenceZNode object.